python-package/xgboost/testing/updater.py

```diff
@@ -218,8 +218,13 @@ def check_extmem_qdm(
         )
 
     booster_it = xgb.train({"device": device}, Xy_it, num_boost_round=8)
-    X, y, w = it.as_arrays()
-    Xy = xgb.QuantileDMatrix(X, y, weight=w)
+    it = tm.IteratorForTest(
+        *tm.make_batches(
+            n_samples_per_batch, n_features, n_batches, use_cupy=device != "cpu"
+        ),
+        cache=None,
```

Comment:
```suggestion
        cache="cache",
        on_host=on_host,
```
Did you mean to re-create the same iterator?

Reply:
They are not the same, the second one doesn't cache data to the disk.